### DISCUSSION TITLE
Heroku deploy - files are overwritten

### DISCUSSION THREAD

**Discussion by dominikjasek** [strapi/strapi#7522] [Legacy Topics]
Hey, I have following problem:

1) I deploy my project to Heroku using `git push heroku master`
2) On Heroku server, I upload some files
3) I need to make some changes in backend strapi - so I do `git push heroku master` again
4) The files I uploaded previously get overwritten and are not available anymore

Is there existing on heroku anything like `pull `or `merge`?

Thank you

**Comment by derrickmehaffy** [comment]
Hello! Please see the following FAQ on the docs: https://strapi.io/documentation/v3.x/getting-started/troubleshooting.html#why-is-my-app-s-database-and-uploads-resetting-on-heroku

The short answer is you need to use an upload provider due to how Heroku's dyno works, there is no other option in this case as it's not directly an issue with Strapi.

**Reply by dominikjasek** [reply]
Thanks, which upload provider would you personally recommend to me? I am using mongoDB...

**Reply by derrickmehaffy** [reply]
That largely depends on you and comparing the cost between them. To my knowledge there is no real "free" option

**Reply by dominikjasek** [reply]
Thanks, I got it
